tests/ha/test_control_pod_restart.py

```diff
@@ -0,0 +1,417 @@
+        cls.rest_iam_user = RestIamUser()
+
+        for node in range(cls.num_nodes):
+            cls.host = CMN_CFG["nodes"][node]["hostname"]
+            cls.username.append(CMN_CFG["nodes"][node]["username"])
+            cls.password.append(CMN_CFG["nodes"][node]["password"])
+            if CMN_CFG["nodes"][node]["node_type"] == "master":
+                cls.host_master_list.append(cls.host)
+                cls.node_master_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+                cls.hlth_master_list.append(Health(hostname=cls.host,
+                                                   username=cls.username[node],
+                                                   password=cls.password[node]))
+            else:
+                cls.host_worker_list.append(cls.host)
+                cls.node_worker_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+
+        cls.rest_obj = S3AccountOperations()
+        cls.s3_mp_test_obj = S3MultipartTestLib(endpoint_url=S3_CFG["s3_url"])
+        cls.test_file = "ha-mp_obj"
+        cls.test_dir_path = os.path.join(TEST_DATA_FOLDER, "HATestMultipartUpload")
+        control_pods = cls.node_master_list[0].get_pods_node_fqdn(const.CONTROL_POD_NAME_PREFIX)
+        ctrl_pod = list(control_pods.keys())[0]
+        backup_path = cls.node_master_list[0].backup_deployment(
+            deployment_name=cls.node_master_list[0].get_deploy_replicaset(ctrl_pod)[1])[1]
+        cls.original_backup = backup_path.split(".")[0] + "_original.yaml"
+        cls.node_master_list[0].rename_file(old_filename=backup_path,
+                                            new_filename=cls.original_backup)
+        cls.original_control_node = control_pods.get(ctrl_pod)
+        cls.multipart_obj_path = None
+
+    def setup_method(self):
+        """
+        This function will be invoked prior to each test case.
+        """
+        LOGGER.info("STARTED: Setup Operations")
+        self.random_time = int(time.time())
+        self.restore_node = False
+        self.restore_ip = False
+        self.deploy = False
+        self.s3_clean = {}
+        LOGGER.info("Check the overall status of the cluster.")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        if not resp[0]:
+            resp = self.ha_obj.restart_cluster(self.node_master_list[0])
+            assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Cluster status is online.")
+        self.s3acc_name = "{}_{}".format("ha_s3acc", int(perf_counter_ns()))
+        self.s3acc_email = "{}@seagate.com".format(self.s3acc_name)
+        self.bucket_name = "ha-mp-bkt-{}".format(self.random_time)
```

Comment:
can we only go with int(perf_counter_ns())

Reply:
Not needed, as long as we are getting randomness, either should work.